### PR TITLE
Statemachine, task refactor

### PR DESCRIPTION
**Pull request by joelrebel**
#### What does this PR do

Prep work to support task restarts, inband firmware installs.

- Moves away from the [stateswitch](https://github.com/filanov/stateswitch) library, to internal packages.
- Refactors `Task` objects for KV storage.
- Adds `Steps` as the smallest unit of work in a `Task` `Action`. 
- Use `github.com/stretchr/testify/mock` instead of `go.uber.org/mock/gomock`.
- Updates `export` CLI command for current Flasher flow diagram.

The commits in this PR are grouped by the kind of changes, recommend reviewing in commit order.

### CODE REVIEW

File: internal/worker/worker.go [hunk 1]
```diff
@@ -101,6 +98,11 @@ func (h *ConditionTaskHandler) Handle(ctx context.Context, condition *rctypes.Co
 		return controller.ErrRetryHandler
 	}
 
+	// TODO: encrypt BMC credentials before store in prod
```
Comment (DoctorVin) [inline internal/worker/worker.go:101]: We can't leave that as a `TODO`. Flasher should not encrypt anything on its own for any storage. It should always get credentials for the BMC from FleetDB.

**Review by DoctorVin — CHANGES_REQUESTED** [review]
I was cool with everything until I got to "Flasher needs to encrypt BMC credentials." I think giving Flasher that responsibility is outside the scope of its work, and it will be a red flag for a lot of folks.

File: internal/model/publisher.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	Publish(ctx context.Context, task *Task)
+}
+
+// StatusPublisher implements the Publisher interface
+// to wrap the condition controller publish method
+type StatusPublisher struct {
+	cp controller.ConditionStatusPublisher
+}
+
+// TODO: this needs to publish the whole task to the KV
```
[the remark this answers is not in the document]
Comment (joelrebel) [reply internal/model/publisher.go:20]: yep, that would be the next round of work

File: internal/install/task_handler.go [hunk 3]
```diff
@@ -66,209 +64,91 @@ func (t *handler) Query(ctx context.Context) error {
...
-
-		// Final is set to true when its the last action in the list.
-		Final: true,
-
-		BMCResetPreInstall:       t.bmcResetPreInstall,
-		BMCResetPostInstall:      bmcResetPostInstall,
-		BMCResetOnInstallFailure: bmcResetOnInstallFailure,
+		return err
 	}
+	// Setting this causes the action SM to not download the file
```
[the remark this answers is not in the document]
Comment (joelrebel) [reply internal/install/task_handler.go:88]: yeah, this is when `flasher install` is invoked with, 
```
flasher install  ... --file blob.bin 
```

File: internal/model/task.go [hunk 4]
```diff
@@ -34,128 +35,97 @@ const (
...
-type Actions []*Action
-
-// ByID returns the Action matched by the identifier
-func (a Actions) ByID(id string) *Action {
-	for _, action := range a {
-		if action.ID == id {
-			return action
-		}
-	}
+	TaskVersion = "0.1"
```
[the remark this answers is not in the document]
Comment (joelrebel) [reply internal/model/task.go:43]: I though `0.1` is valid semver?

File: internal/install/task_handler.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (DoctorVin) [reply internal/install/task_handler.go:88]: I guess my question is: where does that blob come from? If this is an inband thing, then the out-of-band partner downloads it? Or does this get shipped through NATS? If the latter, I think we'll have problems due to the lack of storage we have in the hollow NATS cluster.

File: internal/model/task.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (DoctorVin) [reply internal/model/task.go:43]: It is, but it's also pre-release.

File: internal/worker/worker.go (hunk 1 above)
Comment (DoctorVin) [reply internal/worker/worker.go:101]: Can you kill that `TODO` line then please? Just so any folks who might be snooping on our code don't get a Wrong Idea? tyvm.

File: internal/model/firmware.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"strings"
+
+	"github.com/bmc-toolbox/common"
+)
+
+// Firmware includes a firmware version attributes and is part of FirmwareConfig
+//
+// nolint:govet // fieldalignment struct is easier to read in the current format
```
[the remark this answers is not in the document]
Comment (joelrebel) [reply internal/model/firmware.go:11]: We can do that once its moved to the shared CI linter configuration.

File: internal/install/task_handler.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (joelrebel) [reply internal/install/task_handler.go:88]: The `flasher install` command is meant to be invoked by an operator manually. Its used to test/install updates without the whole NATS shebang.

File: internal/model/task.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (joelrebel) [reply internal/model/task.go:43]: ah, now I get what you mean there, I tend to be conservative with versioning, will bump to `1.0` instead.

**Comment by joelrebel** [comment]
rebase on main

**Review by DoctorVin — DISMISSED** [review]

**Comment by joelrebel** [comment]
Since I cherry-picked commits from main in the previous rebase attempt, the conflicts with main couldn't be resolved.
This force push includes a proper rebase on main.

**Pull request merged**